native/acter/src/api/room.rs

```diff
@@ -961,196 +948,149 @@ impl Room {
             .await?
     }
 
-    pub async fn read_receipt(&self, event_id: String) -> Result<bool> {
+    pub async fn image_binary(&self, event_id: String) -> Result<FfiBuffer<u8>> {
```

Comment:
these three functions basically do the same just match for different message-types, no? Can't we just have a single `message_binary` that simply fetches internally?

Reply:
I have no ideas about how can I abstract these functions into single.
![image](https://github.com/acterglobal/a3/assets/48786056/a9cfa928-a754-4d0f-b47b-44073c5d8f77)

Reply:
well, the input and output are the same types, so we can just collapse everything around and make an internal `match` rather than a `let`. something along the lines of:

```rust
  match &m.content.msgtype {
     MessageType::Image(content) => {
       // do what we do with the image case
       //..
       return Ok(FfiBuffer::new(data));
     },
     MessageType::Audio(content => {
       // do what we do with the image case
       //..
       return Ok(FfiBuffer::new(data));
     }
     // ... and other cases
     // ..
     _ => { bail!("Invalid file format"); }
     }
  }
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1175/commits/02b4aea043cc6cfe010f2f0789059630f5fdd220